### DISCUSSION TITLE
deepgram live steaming help

### DISCUSSION THREAD

**Discussion by Andy1996247** [deepgram/community#573] [General help] [answered]
using this giude https://developers.deepgram.com/docs/getting-started-with-live-streaming-audio and this guide https://www.twilio.com/docs/voice/tutorials/consume-real-time-media-stream-using-websockets-python-and-flask and this giude https://elevenlabs.io/docs/api-reference/websockets  im trying to make a python code that can talk  to https://github.com/filip-michalsky/SalesGPT from my phone. it will be able to steam the data  to Twilio to handle the phone call, then twilio steams the data to deegram to generate the voice output to text to send to the run_api.py for SalesGPT then use ElevenLabs to text to Speech the text input from SalesGPT and send it back to the caller.  

this my code so far 

from flask import Flask, request
from twilio.twiml.voice_response import VoiceResponse
import websocket
import deepgram
import requests

# Initialize Flask app
app = Flask(__name__)

# Initialize Deepgram client
client = deepgram.Client("<YOUR_DEEPGRAM_API_KEY>")

@app.route("/voice", methods=['GET', 'POST'])
def voice():
    # Start the Twilio voice response
    response = VoiceResponse()

    # Stream the call data to Deepgram for voice-to-text
    result = client.transcription.streaming(iter(request.data))

    # Send the transcribed text to SalesGPT
    sales_gpt_response = requests.post("http://localhost:5000/generate", json={"text": result['transcript']})

    # Use ElevenLabs to convert SalesGPT response to speech
    ws = websocket.create_connection("wss://api.elevenlabs.io/synthesize")
    ws.send(sales_gpt_response.text)
    audio_data = ws.recv()

    # Stream the audio data back to the caller
    response.play(audio_data)

    return str(response)

if __name__ == "__main__":
    app.run(debug=True)

im gett this the error 

Traceback (most recent call last):
  File "/home/grandview/Documents/WorkSpaces/SalesGPT/try3.py", line 12, in <module>
    client = deepgram.Client("<YOUR_DEEPGRAM_API_KEY>")
AttributeError: module 'deepgram' has no attribute 'Client'. Did you mean: 'client'?

am i Initializeing Deepgram correctly or is the code going to work? i have tried to export my Deegram api key plus add it the .env file. 

any help would be appreciated. im new to codeing

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
Hi @Andy1996247 

This is code from `v2` of the Python SDK.
```
client = deepgram.Client("<YOUR_DEEPGRAM_API_KEY>")
```

            To fix this issue:
                1. You need to revert to the previous version 2 of the SDK: pip install deepgram-sdk==2.12.0
                2. or, update your application's code to use version 3 of this SDK. See the README for more information.

            Things to consider:

                - This Version 3 of the SDK requires Python 3.10 or higher.
                  Older versions (3.9 and lower) of Python are nearing end-of-life: https://devguide.python.org/versions/
                  Understand the risks of using a version of Python nearing EOL.

                - Version 2 of the SDK will receive maintenance updates in the form of security fixes only.
                  No new features will be added to version 2 of the SDK.
